frontend/src/campaign/speech.js

```diff
@@ -126,23 +127,61 @@ export class Speech extends React.Component {
         }
     }
 
+    makeProposalDict(speech) {
+        const dict = {
+            'low': 0, 'medium': 0, 'high': 0,
+        };
+        for (const topic of Object.keys(speech)) {
+            if (speech[topic] === 1) {
+                dict['low'] += 1;
+            } else if (speech[topic] === 3) {
+                dict['medium'] += 1;
+            } else if (speech[topic] === 5) {
+                dict['high'] += 1;
+            }
+        }
+        return dict;
+    }
+
+    noProblem(dict) {
+        let acceptable = true;
+        let atMaxStatement = '';
+        for (const priority of Object.keys(this.max_priority_points)) {
+            if (dict[priority] > this.max_priority_points[priority]) {
+                acceptable = false;
+            } else {
+                atMaxStatement += 'You can have ' + (this.max_priority_points[priority]
```

Comment:
I'd suggest factoring the calculation out and using a template literal for ease of reading:

```
const priority_points_left = this.max_priority_points[priority] - dict[priority]
atMaxStatement += `You can have ${priority_points_left} more sectors at ${priority} priority.\n';
```